pkg/controller/nodeipam/ipam/multi_subnetworks_handler_test.go

```diff
@@ -0,0 +1,272 @@
+					},
+				},
+			},
+			wantFound:  false,
+			wantSubnet: "",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			gotFound, gotSubnet := getNodeSubnetLabel(tc.node)
+			if gotFound != tc.wantFound {
+				t.Errorf("getNodeSubnetLabel() gotFound = %v, want %v", gotFound, tc.wantFound)
+			}
+			if gotSubnet != tc.wantSubnet {
+				t.Errorf("getNodeSubnetLabel() gotSubnet = %v, want %v", gotSubnet, tc.wantSubnet)
+			}
+		})
+	}
+}
+
+func TestGetSubnetWithPrefixFromURL(t *testing.T) {
+
+	tests := []struct {
+		name             string
+		url              string
+		wantSubnetName   string
+		wantSubnetPrefix string
+		wantErr          bool
+	}{
+		{
+			name:             "valid subnet URL",
+			url:              "https://www.googleapis.com/compute/v1/projects/my-project/regions/us-central1/subnetworks/my-subnet",
+			wantSubnetName:   "my-subnet",
+			wantSubnetPrefix: "projects/my-project/regions/us-central1/subnetworks/",
+			wantErr:          false,
+		},
+		{
+			name:             "URL without https prefix",
+			url:              "projects/another-project/regions/europe-west1/subnetworks/another-subnet",
+			wantSubnetName:   "another-subnet",
+			wantSubnetPrefix: "projects/another-project/regions/europe-west1/subnetworks/",
+			wantErr:          false,
+		},
+		{
+			name:    "URL missing 'projects/'",
+			url:     "https://www.googleapis.com/compute/v1/my-project/regions/us-central1/subnetworks/my-subnet",
+			wantErr: true,
+		},
+		{
+			name:    "Incorrect URL path, too short",
+			url:     "projects",
+			wantErr: true,
+		},
+		{
+			name:    "URL with 'subnetworks/' but no project parts",
+			url:     "https://www.googleapis.com/compute/v1/regions/us-central1/subnetworks/my-subnet",
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			gotSubnetName, gotSubnetPrefix, err := getSubnetWithPrefixFromURL(tc.url)
+			if (err != nil) != tc.wantErr {
+				t.Errorf("getSubnetWithPrefixFromURL() error = %v, wantErr %v", err, tc.wantErr)
+				return
+			}
+			if gotSubnetName != tc.wantSubnetName {
+				t.Errorf("getSubnetWithPrefixFromURL() gotSubnetName = %v, want %v", gotSubnetName, tc.wantSubnetName)
+			}
+			if gotSubnetPrefix != tc.wantSubnetPrefix {
+				t.Errorf("getSubnetWithPrefixFromURL() gotSubnetPrefix = %v, want %v", gotSubnetPrefix, tc.wantSubnetPrefix)
+			}
+
+		})
+	}
+}
+
+func TestUpdateNodeTopology(t *testing.T) {
+
+	testClusterValues := gce.DefaultTestClusterValues()
+	testClusterValues.SubnetworkURL = "https://www.googleapis.com/compute/v1/projects/my-project/regions/us-central1/subnetworks/subnet-def"
+	fakeGCE := gce.NewFakeGCECloud(testClusterValues)
+
+	emptyNodeTopologyCR := &ntv1.NodeTopology{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+		},
+		// Status is initially left empty
+	}
+	ntClient := ntfakeclient.NewSimpleClientset(emptyNodeTopologyCR)
+
+	tests := []struct {
```

Comment:
can we also verify the subnet path? This is useful when we support subnets from different projects in the future.